io_gripper_controller/src/io_gripper_controller.cpp

```diff
@@ -0,0 +1,1379 @@
+        }
+      }
+      reconfigure_state_buffer_.writeFromNonRT(reconfigure_state_type::CHECK_STATE);
+      break;
+
+    case reconfigure_state_type::CHECK_STATE:
+      check_state_ios_ = false;
+      for (const auto & io : conf_it_.state_high)
+      {
+        setResult = find_and_get_state(io, state_value_);
+        if (!setResult)
+        {
+          check_state_ios_ = false;
+          RCLCPP_ERROR(get_node()->get_logger(), "Failed to get the state for %s", io.c_str());
+        }
+        else
+        {
+          if (!(std::abs(state_value_ - 1.0) < std::numeric_limits<double>::epsilon()))
+          {
+            check_state_ios_ = false;
+            RCLCPP_ERROR(get_node()->get_logger(), "value for state doesn't match %s", io.c_str());
+            break;
+          }
+          else
+          {
+            check_state_ios_ = true;
+          }
+        }
+      }
+
+      for (const auto & io : conf_it_.state_low)
+      {
+        setResult = find_and_get_state(io, state_value_);
+        if (!setResult)
+        {
+          RCLCPP_ERROR(get_node()->get_logger(), "Failed to get the state for %s", io.c_str());
+        }
+        else
+        {
+          if (!(std::abs(state_value_ - 0.0) < std::numeric_limits<double>::epsilon()))
+          {
+            RCLCPP_ERROR(get_node()->get_logger(), "value doesn't match %s", io.c_str());
+            check_state_ios_ = false;
+            break;
+          }
+          else
+          {
+            check_state_ios_ = true;
+          }
+        }
+      }
+
+      if (check_state_ios_)
+      {
+        for (size_t i = 0; i < conf_it_.joint_states.size(); ++i)
+        {
+          joint_state_values_[i + params_.open_close_joints.size()] = conf_it_.joint_states[i];
+        }
+        reconfigure_state_buffer_.writeFromNonRT(reconfigure_state_type::IDLE);
+        configure_gripper_buffer_.writeFromNonRT("");
+        reconfigureFlag_.store(false);
+      }
+      break;
+    default:
+      break;
+  }
+}
+
+controller_interface::CallbackReturn IOGripperController::check_parameters()
+{
+  if (params_.open_close_joints.empty())
+  {
+    RCLCPP_FATAL(get_node()->get_logger(), "Size of open close joints parameter cannot be zero.");
+    return CallbackReturn::FAILURE;
+  }
+
+  if (params_.open.joint_states.empty())
+  {
+    RCLCPP_FATAL(get_node()->get_logger(), "Size of joint states parameters cannot be zero.");
+    return CallbackReturn::FAILURE;
+  }
+
+  if (params_.open.command.high.empty() and params_.open.command.low.empty())
+  {
+    RCLCPP_FATAL(
+      get_node()->get_logger(), "Size of open command high and low parameters cannot be zero.");
+    return CallbackReturn::FAILURE;
+  }
+
+  if (params_.open.state.high.empty() and params_.open.state.low.empty())
+  {
+    RCLCPP_FATAL(
+      get_node()->get_logger(), "Size of open state high and low parameters cannot be zero.");
+    return CallbackReturn::FAILURE;
+  }
+
+  if (
```

Comment:
This should be optional.

Reply:
I will update the required code for the optional. Thanks.